### DISCUSSION TITLE
EN 11.7: Dark and light mode requirements

### DISCUSSION THREAD

**Discussion by erikgustafsson** [Nordic-Accessibility-Community-Group/en-301-549-resources-and-eaa-monitoring#6] [General]
What is the view of monitoring agencies for dark mode and light mode in the different countries?

For Sweden the Swedish monitoring agency for the public sector, [Digg states Dark mode is a requirement.](https://www.digg.se/webbriktlinjer/alla-webbriktlinjer/respektera-anvandarens-installningar).

PTS will follow the same ruling for the private sector (EAA).

**Comment by christerFunka** [comment]
How should the criterion be tested? Is it enough to change the appearance to dark in the Firefox and Chrome settings or is it the operating system's contrast mode that applies? 
Will it fail if it is only a single component where the setting does not apply?

**Comment by sander-nl** [comment]
There is also an interesting discussion on this topic in the EN standard: https://labs.etsi.org/rep/HF/en301549/-/issues/23.

**Reply by plannero** [reply]
It would be very good if members of the Nordic Accessibility Community Group could propose a sustainable formulation for a future 11.7 (or possibly one for web content in a new 9.7 clause and another for apps in 11.7). 

As you can see (if you have time) in the thread Sander was linking to, there are many challenges. You do not need to address all of them at once, but maybe focus on dark/light mode: 

Should or should it not be a requirement for websites to provide both a darkmode and a lightmode css? And what about apps? 

Is such a requirement important enough from an accessibility point of view to motivate the challenges associated with producing and maintaining more than one design? Please take into account 
* The popularity of darkmode, 
* The cost of developing, testing and maintaining another design, 
* The challenges involved in publishing images when they will sometimes appear in a light context and sometimes in a dark context, 
* The needs of persons with disabilities, 
* The advances of AI (that may soon be able to automatically render a reasonably accessible dark/lightmode based on user preferences regardless of website css).

**Reply by sander-nl** [reply]
I will try to formulate some thoughts, but until then I'd like to mention something related to Pär's last bullet point. Chrome and Edge DevTools already have an option to enable automatic dark mode. This can be found under the "Rendering" tab. This doesn't work perfectly though.

**Reply by plannero** [reply]
In the ETSI system, the latest comments indicate that the dark/lightmode setting, after all, is not an accessibility setting.  Do you agree? (In Windows and MacOS/iOS the color theme setting is not to be found under Accessibility, but under Android it is.)

**Reply by christerFunka** [reply]
Agree that it is not an accessibility setting. That would be the same as saying that the volume buttons belong to accessibility. 
I have an Android OnePlus 10 Pro and the dark/light settings are not found under “Accessibility Settings” but under “Display & Brightness”. On an Android OnePlus 10 Pro, I can't change fonts nowadays either.

I get the feeling that when people talk about Android they are talking about Samsung.

**Reply by plannero** [reply]
Thanks for the comment about Samsung-Android!

And thanks for the analogy about volume buttons! Analogies can really help thinking! 
In fact, there are several accessibility criteria that relate to volume buttons. Eg. 5.1.3.11-5.1.3.13. Those are usually about volume buttons for speech output, but not only. 

I found some statistics regarding light sensitivity over at https://www.theraspecs.com/blog/photophobia-light-sensitivity-facts-and-statistics/ - it seems many different diagnoses are related to this need. But still difficult to estimate the need for darkmode..

**Reply by sander-nl** [reply]
I tend to agree that dark and light mode is not an accessibility setting. Especially after reading [Dark mode & accessibility myth](https://stephaniewalter.design/blog/dark-mode-accessibility-myth-debunked/). For example, for some people it is even harder to read light text on a dark background. 

The article also lists some interesting 'work-arounds' in the browser (i.e. reader mode in Firefox). What if a website is coded correctly so that (for example) Firefox's reader mode can present it properly and in the colors the user wants (not only light or dark, but also sepia or custom), does the website comply to 11.7 then? I tend to think so, because there is a way to respect the user preferences. 

An other consideration: if light and dark modes are required, where is the threshold? When is it 'dark mode enough'? I don't think it is easy or even possible to define this. 

By the way: I really like dark mode on apps and sites and use it where I can. It is another thing to make this a requirement.

**Reply by plannero** [reply]
Thanks @sander-nl !
I agree that the existence of Reader Mode is a valid argument against making light+darkmode a requirement.
But I do not agree that some people's need to have a light mode makes it a non-a11y setting. If people find dark-mode difficult to read, then the option to always have a light mode alternative increases accessibility.
As for myself, I tend to prefer light mode, but I agree that one or a few individuals' preferences should not affect what is made a requirement.

**Reply by pbb72** [reply]
Dark mode in itself is not "more accessible". This is where the "Dark mode & accessibility myth" article falls flat on it's face. The accessible aspect is the ***choice*** of having either dark mode or light mode. The same goes for high contrast mode; it's not more accessible by rule if a site uses high contrast colors, but offering the option (or better, following a choice already made at a higher level) makes the site more accessible.

Reader mode is an interesting aspect. Obviously, this means having to test that your site works properly in Reader mode. But can you require that a user with an accessibility need uses a browser that has a Reader mode? Said in another way, am I compliant if certain accessibility aspects only work in specific browsers?

Back to the original question, yes I feel websites should offer the option to follow user settings. This goes for light/dark mode, high/low contrast mode, font size and face, reduced motion, and any other user-setting that is visible to the site. But I don't feel it has to be the default setting, I find it better if there was a (single) setting to switch between designer choice and user choice.
